content/setup/remotes/bitbucketserver.md

```diff
@@ -0,0 +1,80 @@
++++
+date = "2016-04-20T16:00:21-08:00"
+draft = false
+title = "Bitbucket Server(Stash)"
+weight = 1
+menu = "remotes"
+toc = true
++++
+
+Drone comes with built-in support for Bitbucket Server(stash) To enable Bitbucket Server you should configure the Bitbucket driver using the following environment variables:
+
+```bash
+REMOTE_DRIVER=bitbucketserver
+REMOTE_CONFIG=https://{servername}?consumer_key={consumer_key}&git_username={git_username}&git_password={git_password}&consumer_rsa={path_private_key}&open={unused}
+```
+
+# Bitbucket Server configuration
+
+The following is the standard URI connection scheme:
+
+```
+scheme://host[?options]
+```
+
+The components of this string are:
+
+* `scheme` server protocol `http` or `https`.
+* `host` server address to connect to.
+* `?options` connection specific options.
+
+# Bitbucket Server options
+
+This section lists all connection options used in the connection string format. Connection options are pairs in the following form: `name=value`. The value is always case sensitive. Separate options with the ampersand (i.e. &) character:
+
+* `consumer_key` oauth client id for registered application (must add in application links incoming auth in Bitbucket server).
+* `git_username` Service account that can http clone the projects.
+* `git_password` password for the service account that can http clone the projects.
+* `consumer_rsa` Path to the private key for oauth1 (will need to copy into a customer docker image or map the volume to the correct location).
+* `open=false` not currently used but will be added later
+
+# Why do we need the service account?
+
+Unforunately Bitbucket server doesn't support oauth http cloning of repos. This means we need a service account that can clone repos for us. Please please go [here](https://jira.atlassian.com/browse/BSERV-2722) to vote for this feature and we can move to not need service accounts
+
+
+# Bitbucket Server registration
+
+You must register your application with Bitbucket in order to allow for auth1. Navigate to admin choose application links and add a new application link
+
+Please use `http://drone.mycompany.com/authorize` as the Authorization callback URL in the incoming auth(at the bottom):
+
+* Add the url to drone on the application path
+* You'll also need to add the public key (public key to the private key in consumer_rsa_key file)
+
```

Comment:
can you add a note that this will need to be mounted as a volume into the container as part of your `docker run` command? and provide an example code snippet?

---

The oauth consumer private key needs to be mounted into the Docker container. You can mount the key into the container using the `--volume` command line parameter:
```
docker run --volume=/host/path/to/key.pem:/container/path/to/key.pem
```

The path to your oauth consumer private key should be defined in your configuration:

```
consumer_rsa=/container/path/to/key.pem
```